pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -75,11 +75,16 @@ func Add(mgr manager.Manager) error {
 }
 
 func (c *configResource) Process() (reconcile.Result, error) {
-	// Don't update the ConfigMap if an update is required
+	// Don't update the ConfigMap if an update is about to be scheduled
 	if req, err := ispnctrl.IsUpgradeRequired(c.infinispan); req || err != nil {
 		return reconcile.Result{RequeueAfter: consts.DefaultWaitOnCreateResource}, nil
 	}
 
+	// Don't update the configMap if an update is in progress
+	if c.infinispan.IsConditionTrue(ispnv1.ConditionUpgrade) {
```

Comment:
Naming could be confusing. I'd expect that ConditionUpgrade(ing) is true till the cluster is back up and everything was successfully updated. ClusterStopped condition would be better in this place.

Reply:
> Naming could be confusing. I'd expect that ConditionUpgrade(ing) is true till the cluster is back up and everything was successfully updated.

`ConditionUpgrade` is true until the original StatefulSet has been deleted, then it is set to `false` and the reconcile request requeued in order to recreate the new StatefulSet. I don't think we should change the name or semantics of this condition at this stage, as users could be depending on this logic via `kubectl  wait --for condition=`.

Reply:
>  I don't think we should change the name or semantics of this condition at this stage

I agree. Just wanted to point out that instinctively I'd expect the condition to behave differently. Upgrade isn't really finished by deleting resources.